srv/modules/runners/validate.py

```diff
@@ -771,66 +792,78 @@ def ceph_version(self):
         """
         Scan all minions for ceph versions in their repos.
         """
-        local = salt.client.LocalClient()
-        contents = local.cmd(self.search,
-                             'pkg.latest_version',
-                             ['ceph-common'],
-                             tgt_type="compound")
-        for minion, version in contents.items():
-            # sometimes, version contains a string value
-            # other times, it contains an empty dict
-            log.debug("VALIDATE ceph_version: minion ->{}<- latest_version version ->{}<-"
-                      .format(minion, version))
-            if not version:
-                info = local.cmd(minion, 'pkg.info_installed', ['ceph-common'])
-                if info and info[minion] and 'version' in info[minion]['ceph-common']:
-                    version = info[minion]['ceph-common']['version']
-                    log.debug("VALIDATE ceph_version: minion ->{}<- info_installed version ->{}<-"
-                              .format(minion, version))
+        self._check_installed()
+        self._check_available()
+        self._set_pass_status('ceph_version')
+
+    def _check_installed(self):
+        """
+        Check for installed Ceph packages.  The query is faster and a fresh
+        install only happens once.
+        """
+        search = __utils__['deepsea_minions.show']()
+        results = self._silent_search(search, 'pkg.info_installed')
+
+        for minion in results:
+            if isinstance(results[minion], dict) and self.package in results[minion]:
+                if 'version' in results[minion][self.package]:
+                    version = self._check_version(minion, 'pkg.info_installed',
+                                                  results[minion][self.package]['version'])
```

Comment:
if version is set to `""`, which happens in case of an error in the retrieved data see `check_version`

``` python
if version.split('.')[0].isdigit():
    return version
log.debug("VALIDATE {} on minion {} returns {}".format(func, minion, version))
    return ""
```

The comparison of `LooseVersion`s will fail.